src/engraving/libmscore/beam.cpp

```diff
@@ -267,9 +254,7 @@ void Beam::draw(mu::draw::Painter* painter) const
 
 bool Beam::isNoSlope() const
 {
-    PointF currentBeamPos = beamPos();
-
-    return qFuzzyCompare(currentBeamPos.x(), currentBeamPos.y());
+    return _userModified[0] && qFuzzyCompare(beamPos().first, beamPos().second);
```

Comment:
the key fix - without checking _userModified[0] it was assuming beams must be always stay horizontal just because they had been previously